codegen/smithy-ruby-codegen/src/main/java/software/amazon/smithy/ruby/codegen/middleware/MiddlewareBuilder.java

```diff
@@ -191,6 +180,63 @@ public void addDefaultMiddleware(GenerationContext context) {
                         "@stubs = Hearth::Stubbing::Stubs.new")
                 .build();
 
+        ClientConfig maxAttempts = (new ClientConfig.Builder())
+                .name("max_attempts")
+                .type("Integer")
+                .defaultValue("3")
+                .documentation(
+                        "An integer representing the maximum number of attempts that will be made for a "
+                        + "single request, including the initial attempt."
+                )
+                .build();
+
+        ClientConfig retryMode = (new ClientConfig.Builder())
+                .name("retry_mode")
+                .type("String")
+                .defaultValue("'standard'")
+                .documentation(
+                        "Specifies which retry algorithm to use. Values are: \n"
+                        + " * `standard` - A standardized set of retry rules across the AWS SDKs. This includes support"
+                        + " for retry quotas, which limit the number of unsuccessful retries a client can make.\n"
+                        + " * `adaptive` - An experimental retry mode that includes all the"
+                        + " functionality of `standard` mode along with automatic client side"
+                        + " throttling.  This is a provisional mode that may change behavior"
+                        + " in the future."
+                )
+                .build();
+
+        ClientConfig adaptiveRetryWaitToFill = (new ClientConfig.Builder())
+                .name("adaptive_retry_wait_to_fill")
+                .type("Boolean")
+                .defaultValue("true")
+                .documentation(
+                       "Used only in `adaptive` retry mode. When true, the request will sleep until there is"
+                       + " sufficient client side capacity to retry the request. When false, the request will"
+                       + " raise a `CapacityNotAvailableError` and will not retry instead of sleeping."
+                )
+                .build();
+
+        ClientConfig retryQuota = (new ClientConfig.Builder())
+                .name("retry_quota")
+                .initializationCustomization("@retry_quota = Hearth::Retry::RetryQuota.new")
+                .build();
+
+        ClientConfig clientRateLimiter = (new ClientConfig.Builder())
+                .name("client_rate_limiter")
+                .initializationCustomization("@client_rate_limiter = Hearth::Retry::ClientRateLimiter.new")
+                .build();
+
+        Middleware retry = (new Middleware.Builder())
+                .klass("Hearth::Middleware::Retry")
+                .step(MiddlewareStackStep.RETRY)
+                .addConfig(maxAttempts)
+                .addConfig(retryMode)
+                .addConfig(adaptiveRetryWaitToFill)
+                .addParam("error_inspector_class", "Hearth::Retry::ErrorInspector")
+                .addConfig(retryQuota)
```

Comment:
I think these should be params rather than Config, because we dont want to do an options.fetch on them

Reply:
I don't think that works because I need to do an intializationCustomization